x-pack/platform/plugins/shared/streams/server/lib/content/stream.ts

```diff
@@ -175,3 +172,22 @@ export function resolveAncestors(leafs: string[]) {
   const streamNames = leafs.flatMap((name) => getAncestorsAndSelf(name));
   return uniq(streamNames);
 }
+
+function assertNoConflictingRouting(
+  rootRouting: RoutingDefinition[],
+  importedRouting: RoutingDefinition[]
+) {
+  for (const { destination } of importedRouting) {
+    if (rootRouting.some((rule) => rule.destination === destination)) {
+      throw new ContentPackConflictError(`Child stream [${destination}] already exists`);
```

Comment:
We could allow a merge if it's 100% identical with the existing thing. Think a pure addition to a content pack - if I just added a new stream and didn't change anything else, then it should be fine. What do you think?

Reply:
> We could allow a merge if it's 100% identical with the existing thing

Not only the routing rules but the stream definition and queries / dashboards ?
I shared some thoughts on an upgrade process [here](https://github.com/elastic/streams-program/discussions/428), if that's fine with you I'd like to discuss about this more about this process and tackle this separately